core/node/shared/ids.go

```diff
@@ -43,21 +44,57 @@ func AddressFromSpaceId(spaceId StreamId) (common.Address, error) {
 	return common.BytesToAddress(spaceId[1:21]), nil
 }
 
+func MakeSpaceId() (StreamId, error) {
+	var b [32]byte
+	b[0] = STREAM_SPACE_BIN
+	_, err := rand.Read(b[1:21])
+	if err != nil {
+		return StreamId{}, RiverError(Err_INTERNAL, "failed to create random bytes", "err", err)
+	}
+	return StreamIdFromBytes(b[:])
+}
+
 func MakeChannelId(spaceId StreamId) (StreamId, error) {
 	// replace the first byte with the channel type
 	// copy the 20 bytes of the spaceId address
 	// fill the rest with random bytes
+	b, err := makeSpacePrefixChannelBytes(spaceId)
+	if err != nil {
+		return StreamId{}, err
+	}
+	_, err = rand.Read(b[21:])
+	if err != nil {
+		return StreamId{}, RiverError(Err_INTERNAL, "failed to create random bytes", "err", err)
+	}
+	return StreamIdFromBytes(b[:])
+}
+
+func makeSpacePrefixChannelBytes(spaceId StreamId) (StreamId, error) {
```

Comment:
this is returning a "channel prefix" but the function name makes me think it's returning a space prefix something, it's very unclear - can you simplify this name some more?

Reply:
Oh my bad, I thought I renamed this. Will do!